daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1149,33 +1149,40 @@ private[lf] object SBuiltin {
           }
 
         case None =>
-          def continue(coinst: V.ContractInstance): Control.Expression = {
-            machine.pushKont(KCacheContract(coid))
-            val e = coinst match {
-              case V.ContractInstance(actualTmplId, arg) =>
-                SELet1(
-                  // The call to ToCachedContractDefRef(actualTmplId) will query package
-                  // of actualTmplId if not known.
-                  SEVal(ToCachedContractDefRef(actualTmplId)),
-                  SELet1(
-                    SEImportValue(Ast.TTyCon(actualTmplId), arg),
-                    SEAppAtomic(SELocS(2), Array(SELocS(1), SEValue(args.get(1)))),
-                  ),
+          machine.disclosedContracts.get(coid) match {
+            case Some(contract) =>
+              machine.addGlobalContract(coid, contract)
+              Control.Value(contract.any)
```

Comment:
Here the disclosed contract are adding to the cached contracts while transaction limits are enforced